packages/image_picker/image_picker_windows/lib/image_picker_windows.dart

```diff
@@ -112,54 +98,81 @@ class ImagePickerWindows extends ImagePickerPlatform {
+      case ImageSource.gallery:
+        const XTypeGroup typeGroup =
+            XTypeGroup(label: 'Images', extensions: imageFormats);
+        final XFile? file = await fileSelector
+            .openFile(acceptedTypeGroups: <XTypeGroup>[typeGroup]);
+        return file;
     }
-    const XTypeGroup typeGroup =
-        XTypeGroup(label: 'images', extensions: imageFormats);
-    final XFile? file = await fileSelector
-        .openFile(acceptedTypeGroups: <XTypeGroup>[typeGroup]);
-    return file;
+    // Ensure that there's a fallback in case a new source is added.
+    // ignore: dead_code
+    throw UnimplementedError('Unknown ImageSource: $source');
   }
 
-  // `preferredCameraDevice` and `maxDuration` arguments are not
-  // supported on Windows. If any of these arguments is supplied,
-  // it'll be silently ignored by the Windows version of the plugin.
-  // `source` is not implemented for `ImageSource.camera` and will
-  // throw an exception.
+  // `preferredCameraDevice` and `maxDuration` arguments are not currently
+  // supported. If any of these arguments is supplied, they will be silently
+  // ignored.
+  //
+  // If source is `ImageSource.camera`, a `StateError` will be thrown
+  // unless a [cameraDelegate] is set.
   @override
   Future<XFile?> getVideo({
     required ImageSource source,
     CameraDevice preferredCameraDevice = CameraDevice.rear,
     Duration? maxDuration,
   }) async {
-    if (source != ImageSource.gallery) {
-      // TODO(azchohfi): Support ImageSource.camera.
-      //                 See https://github.com/flutter/flutter/issues/102115
-      throw UnimplementedError(
-          'ImageSource.gallery is currently the only supported source on Windows');
+    switch (source) {
+      case ImageSource.camera:
+        return super.getVideo(
+            source: source,
+            preferredCameraDevice: preferredCameraDevice,
+            maxDuration: maxDuration);
+      case ImageSource.gallery:
+        const XTypeGroup typeGroup =
+            XTypeGroup(label: 'Videos', extensions: videoFormats);
+        final XFile? file = await fileSelector
+            .openFile(acceptedTypeGroups: <XTypeGroup>[typeGroup]);
+        return file;
     }
-    const XTypeGroup typeGroup =
-        XTypeGroup(label: 'videos', extensions: videoFormats);
-    final XFile? file = await fileSelector
-        .openFile(acceptedTypeGroups: <XTypeGroup>[typeGroup]);
-    return file;
+    // Ensure that there's a fallback in case a new source is added.
+    // ignore: dead_code
+    throw UnimplementedError('Unknown ImageSource: $source');
   }
 
-  // `maxWidth`, `maxHeight`, and `imageQuality` arguments are not
-  // supported on Windows. If any of these arguments is supplied,
-  // it'll be silently ignored by the Windows version of the plugin.
+  // `maxWidth`, `maxHeight`, and `imageQuality` arguments are not currently
+  // supported. If any of these arguments is supplied, they will be silently
+  // ignored.
```

Comment:
They were in https://github.com/flutter/flutter/issues/102320, but since they were optional there I just split out the new https://github.com/flutter/flutter/issues/128453 to continue tracking.

It's not just filtering, the image options resize and re-encode the images to fit. It's really unfortunate; I added some context to the issue.